src/abstractdataframe/join.jl

```diff
@@ -417,6 +426,19 @@ function Base.join(df1::AbstractDataFrame,
     return joined
 end
 
+function Base.join(df1::AbstractDataFrame, df2::AbstractDataFrame,
+                   dfs::AbstractDataFrame...;
+                   on::Union{Symbol, AbstractVector{Symbol}} = Symbol[],
```

Comment:
I guess we should disallow (vectors of) pairs? Also the empty default may give weird results: should it operate on columns that are common to all inputs (which isn't what it does in this PR)?

Reply:
I disallow `Pairs`/vectors of `Pairs` as you propose. Have I missed something?

An empty default to `on` leads to an error unless the join is `kind=:cross` in which case it is not allowed to pass a non-empty `on`.

In general - the behavior of `on` is the same for many as for two data frames - you have to be explicit what you want to join on.